radix-engine-interface/src/blueprints/resource/non_fungible/non_fungible_vault.rs

```diff
@@ -0,0 +1,60 @@
+use crate::blueprints::resource::Proof;
+use crate::*;
+use radix_engine_common::data::scrypto::model::NonFungibleLocalId;
+use radix_engine_interface::blueprints::resource::Bucket;
+use sbor::rust::collections::BTreeSet;
+use sbor::rust::prelude::*;
+
+pub const NON_FUNGIBLE_VAULT_BLUEPRINT: &str = "NonFungibleVault";
+
+pub const NON_FUNGIBLE_VAULT_TAKE_NON_FUNGIBLES_IDENT: &str = "take_non_fungibles";
+
+#[derive(Debug, Clone, Eq, PartialEq, ScryptoSbor)]
+pub struct NonFungibleVaultTakeNonFungiblesInput {
+    pub non_fungible_local_ids: BTreeSet<NonFungibleLocalId>,
+}
+
+pub type NonFungibleVaultTakeNonFungiblesOutput = Bucket;
+
+pub const NON_FUNGIBLE_VAULT_GET_NON_FUNGIBLE_LOCAL_IDS_IDENT: &str = "get_non_fungible_local_ids";
+
+#[derive(Debug, Clone, Eq, PartialEq, ScryptoSbor)]
+pub struct NonFungibleVaultGetNonFungibleLocalIdsInput {}
+
+pub type NonFungibleVaultGetNonFungibleLocalIdsOutput = BTreeSet<NonFungibleLocalId>;
+
+pub const NON_FUNGIBLE_VAULT_RECALL_NON_FUNGIBLES_IDENT: &str = "recall_non_fungibles";
+
+#[derive(Debug, Clone, Eq, PartialEq, ScryptoSbor)]
+pub struct NonFungibleVaultRecallNonFungiblesInput {
+    pub non_fungible_local_ids: BTreeSet<NonFungibleLocalId>,
+}
+
+pub type NonFungibleVaultRecallNonFungiblesOutput = Bucket;
+
+pub const NON_FUNGIBLE_VAULT_CREATE_PROOF_BY_IDS_IDENT: &str = "create_proof_by_ids";
```

Comment:
I have tickets on the work tracker for:
* Rename all `_by_ids` to `non_fungible` in code base
* Rename all proof methods to eg: create_proof_of_all(), create_proof_of_amount()

Is this a good time for that change? Or maybe in a follow-up PR?

Reply:
Yup good point